spec/vector_spec.rb

```diff
@@ -1169,4 +1169,21 @@ def expect_correct_tokens hash
     end
   end
 
+  context "#to_df" do
+    it "converts vector to DataFrame containing that vector" do
+      dv = Daru::Vector.new 1..10, name: :a
+      df = Daru::DataFrame.new({
+        a: 1..10
+      })
+      expect(dv.to_df).to eq(df)
+    end
+  end
+
+  context "#group_by" do
+    it "redirects to DataFrame::group_by" do
+      dv = Daru::Vector.new [1, 1, 2, 2, 3], name: :a
+      expect(dv.group_by([:a]).groups).to eq(dv.to_df.group_by([:a]).groups)
```

Comment:
This test should have the actual result that the call to group_by will generate. Here you're just calling what's inside the `Vector#group_by` method.